CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,184 @@
+
+Some ideas other than the proposed approach include:
+
+1. Exchanges have a delay. Any exchange must be explicitly approved by a governance proposal to be executed.
+   * This relies upon the community as the arbiter of what exchanges are "safe."
+   * This could exacerbate the already-present voter fatigue.
+2. Large-volume exchanges could be made by anyone without Governance approval.
+   * The most compelling approach to this would involve an auction. This is complicated to implement, requires careful design, and was intentionally removed from the original Mento design.
+
+Approach (2) was deemed risky and complicated. Approach (1) felt more in line with the goal of having Granda Mento be simple and safe, but ultimately the involvement of Governance for every exchange felt unsustainable and exhausting for voters. The proposed implementation involving an exchange delay, a multisig for approvals, and Governance in the (likely) rare event the community wishes to veto an exchange, aims to relieve voter exhaustion while still providing the security of community scrutiny.
+
+The proposed implementation prohibits the proposer from cancelling their own exchange proposal after it has been approved. This is to prevent Granda Mento from being used as a "free option," where the proposer can decide against the exchange late in the process if the price of the exchange is no longer favorable.
+
+[Initial discussions around Granda Mento](https://forum.celo.org/t/discussion-on-granda-mento-enabling-larger-stablecoin-mints/966) showed concern around providing a "fair" price to ensure that large players do not have an unfair opportunity to "dump" CELO for stable tokens. Some ideas for pricing were considered:
+
+1. Oracle price at the time of exchange proposal.
+   * This is vulnerable to oracle attacks.
+2. The price is included by the exchange proposer, and is required to be within X% of the current oracle price.
+   * This is vulnerable to oracle attacks.
+   * This puts more control in the hands of the exchange proposer.
+3. Oracle price at the time of exchange execution.
+   * This is vulnerable to oracle attacks.
+   * Exchanger doesn't know the price at the proposal time.
+4. TWAP between exchange proposal and exchange execution.
+   * Involves smart contract changes to have the TWAP available on chain.
+   * Exchanger doesn't know the price at the proposal time.
+
+It's desirable for both the exchanger and for the Celo community to know what price will be used for the exchange-- this way, the exchanger knows what they're committing to, and the community can decide if they agree with the price. Approaches (1) and (2) the only options that involve knowledge of the price at the start of the trade.
```

Comment:
I think it's important to highlight here that you're looking for feedback on 1 vs 2 and it's not been decided yet. This may be the most important point of feedback we're looking for in this CIP as the struct created by the proposeExchange function isn't fully defined until this is decided.

Reply:
From a user need perspective, is (2) more desirable than (1)? If users are indifferent then I think it makes the most sense to go with 1.

Reply:
(1) is the simplest and most analogous to an OTC price quote at time of request. (2) would require more explanation of how the final price within that range is determined.

Reply:
If there's no strong user need for (2) then I'm going to commit to (1)